edb/server/bootstrap.py

```diff
@@ -500,61 +536,137 @@ def prepare_patch(
+            # stdschema
+            delta_command = s_ddl.delta_from_ddl(
+                ddl_cmd, modaliases={}, schema=schema, stdmode=True)
+            schema, _ = _process_delta_params(
+                delta_command, schema, backend_params)
+
+            # We need to extract all ids of new objects created when
+            # applying it to the regular schema, so that we can make sure
+            # to use the same ids in the reflschema.
+            schema_object_ids = _get_schema_object_ids(delta_command)
+
+            # Then apply it to the reflschema, which we will use to drive
+            # the actual table updating.
+            delta_command = s_ddl.delta_from_ddl(
+                ddl_cmd, modaliases={}, schema=reflschema,
+                schema_object_ids=schema_object_ids, stdmode=True)
+            reflschema, plan = _process_delta_params(
+                delta_command, reflschema, backend_params)
+            std_plans.append(delta_command)
+            plan.generate(subblock)
+    else:
+        raise AssertionError(f'unknown patch type {kind}')
+
+    if kind == 'edgeql+schema':
+        # If we are modifying the schema layout, we need to rerun
+        # generate_structure to collect schema changes not reflected
+        # in the public schema and to discover the new introspection
+        # query.
+        reflection = s_refl.generate_structure(
+            reflschema, make_funcs=False,
+        )
+        reflection.intro_schema_delta.dump()
 
-        subblock = current_block.add_block()
+        reflschema, plan = _process_delta_params(
+            reflection.intro_schema_delta, reflschema, backend_params)
+        plan.generate(subblock)
 
         compiler = edbcompiler.new_compiler(
             std_schema=schema,
             reflection_schema=reflschema,
             schema_class_layout=schema_class_layout
         )
 
-        compilerctx = edbcompiler.new_compiler_context(
+        local_intro_sql, global_intro_sql = _compile_intro_queries_stdlib(
+            compiler=compiler,
             user_schema=reflschema,
-            global_schema=schema,
-            bootstrap_mode=True,
+            reflection=reflection,
         )
 
-        for std_plan in std_plans:
-            compiler._compile_schema_storage_in_delta(
-                ctx=compilerctx,
-                delta=std_plan,
-                block=subblock,
+        updates.update(dict(
+            classlayout=reflection.class_layout,
+            local_intro_query=local_intro_sql.encode('utf-8'),
+            global_intro_query=global_intro_sql.encode('utf-8'),
+        ))
+
+        # This part is wildly hinky
+        # We need to delete all the support views and recreate them at the end
+        support_view_commands = metaschema.get_support_views(
+            reflschema, backend_params)
+        for cv in support_view_commands:
+            dv = dbops.DropView(
+                cv.view.name,
+                cascade=True,
```

Comment:
Any way we can avoid `cascade` here?  It's pretty dangerous and can reach into parts that we didn't anticipate, hide bugs etc.

Reply:
For some reason, when I was thinking about this yesterday, figuring out the order to drop the views in seemed like it would be hard.

It obviously was not.